### PR TITLE
new resource "azurerm_data_factory_data_flow"

### LINKED ISSUE #10080: Support for creating Data Factory Data flows
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Currently, I do not see a way to create a Data Flow in Azure Data Factory v2 using this provider. I believe it would make a good addition.

### New or Affected Resource(s)

* azurerm_data_factory

### Potential Terraform Configuration

```hcl
azurerm_data_factory_data_flow { }
```

### PR DESCRIPTION
**Pull request by njuCZ**
fix #10080

doc: https://docs.microsoft.com/en-us/azure/data-factory/concepts-data-flow-overview

image

### CODE REVIEW

File: website/docs/r/data_factory_data_flow.html.markdown [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+* `name` - (Required) The name for the Data Flow Source.
+
+* `description` - (Optional) The description for the Data Flow Source.
+
+* `dataset` - (Optional) A `dataset` block as defined below.
+
+* `linked_service` - (Optional) A `linked_service` block as defined below.
+
+* `schema_linked_service` - (Optional) A `schema_linked_service` block as defined below.
```
Comment (katbyte) [inline website/docs/r/data_factory_data_flow.html.markdown:160]: what are the differences between these two?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: azurerm/internal/services/datafactory/data_factory_data_flow_resource_test.go [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+        "Key1" = "value1"
+      }
+    }
+
+    schema_linked_service {
+      name = azurerm_data_factory_linked_custom_service.test.name
+      parameters = {
+        "Key1" = "value1"
+      }
+    }
```
[the remark this answers is not in the document]
Comment (njuCZ) [reply azurerm/internal/services/datafactory/data_factory_data_flow_resource_test.go:219]: I have asked the service team, they says that `schema_linked_service` is mainly for [Common Data Model](https://docs.microsoft.com/en-us/azure/data-factory/format-common-data-model). it allows the data flow to read schema info from `schema_linked_service` and read data from `linked_service`. In this way data and schema metadata could be placed in different location.

To simplify the acctest, I used the same linkedservice for `linked_service` and `schema_linked_service`. Below are some screenshot of *Common Data Model* in portal:

image

image

File: website/docs/r/data_factory_data_flow.html.markdown (hunk 1 above)
Comment (njuCZ) [reply website/docs/r/data_factory_data_flow.html.markdown:160]: as above

**Comment by njuCZ** [comment]
Hi @katbyte thanks for your review, I have rebased this PR.

**Comment by ms-henglu** [comment]
@katbyte , hi, could you please take another look at this PR? Thanks!

**Review by katbyte — APPROVED** [review]
Thanks @njuCZ - LGTM 🍰

**Pull request merged**